### ISSUE TITLE
ImmutableList.of() --> Collections.emptyList()

### ISSUE THREAD

**Issue by elharo** [prestodb/presto#22835]
Now that we're on JDK 8, Collections.emptyList() is preferable to ImmutableList.of().

Should be a straightforward search and replace followed by import cleanup.

**Comment by jasonmokk** [comment]
Hello, I'd like to work on this issue.

**Comment by ScrapCodes** [comment]
May be I can try this!

**Comment by ScrapCodes** [comment]
```
prashant@prashant:~/work/presto$ find . -type f -exec grep -nHI -e "ImmutableList.of" \{\} + | wc -l
13518
```

**Comment by elharo** [comment]
I don't think that will work since you didn't include the parentheses.

**Comment by ScrapCodes** [comment]
@elharo understood that and corrected myself.

**Comment by tdcmeehan** [comment]
@elharo what's the motivation for this change?

**Comment by elharo** [comment]
tldr; JDK > 3rd party library

https://stackoverflow.com/questions/43291202/collections-emptylist-vs-guavas-immutablelist-of

**Comment by tdcmeehan** [comment]
@elharo from the SO post, I see one answer that lists two reasons:

1) Why introduce a third party dependency when you can use whatever is already built-in to the JDK?
2) The JDK factory uses a dedicated implementation for the empty list, vs. Guava uses an immutable list with no elements.

For point 1, I don't understand what the benefit is, unless we have some sort of overarching goal to remove our Guava dependency.  We already have this dependency because we prefer immutability, so I don't understand why we would want to differentiate empty vs. non-empty immutable collections.

For point 2, I simply don't understand what sort of practical benefit this has.  Can you attempt to explain it?
